### DISCUSSION TITLE
SelectFilter not working in Relationship

### DISCUSSION THREAD

**Discussion by bijoy4067** [filamentphp/filament#10626] [Help] [answered]
hello,
i have NewsResource which has relation with Category and Tag. store and update working well but when i apply SelectFilter::make() from filter it's always return empty. i want its will get category_id = ["1", "2"] and search in category table by using id 1 & 2 and return result.

Here is my code example
Form code sample

Select::make('category_id')
->multiple()
->relationship(name: 'category', titleAttribute: 'title')
->preload(), 

Model RelationShip
public function category()
{
    return $this->belongsTo(Category::class, 'category_id', 'id');
}
its store array in database 

sample data store in category_id column
["1", "2"]

Filter code example
SelectFilter::make('category_id')
->label('category')
->multiple()
->relationship(name: 'searchCategory', titleAttribute: 'title')
->preload()

Model RelationShip
public function searchCategory(): BelongsToMany
{
    return $this->belongsToMany(Category::class, Self::class, 'category_id', 'id');
}

but its return empty
following is output
<img width="1439" alt="image" src="https://github.com/filamentphp/filament/assets/52704882/edef986a-eb2f-45b0-bf1a-5a63343a4cd2">

   Any solution?
Thank Fou For Read.

**Comment by leandrocfe** [comment, marked as answer]
The relationship should be `BelongsToMany` if you are using `multiple()`
https://laravel.com/docs/10.x/eloquent-relationships#many-to-many

**Reply by bijoy4067** [reply]
It working when i change my relation like

public function category(): BelongsToMany
{
    return $this->belongsToMany(Category::class);
}

Thank you so much
